lib/ansible/modules/packaging/os/redhat_repositories.py

```diff
@@ -0,0 +1,400 @@
+                changed=True
+
+            if not current_repos[r]['enabled'] and current_repos[r]['id'] in id:
+                to_enable.append(current_repos[r]['id'])
+                changed=True
+
+        notice("redhat_repository: enable_only: changed=" + str(changed))
+        notice("redhat_repository: enable_only: to_enable=" + str(to_enable))
+        notice("redhat_repository: enable_only: to_disable=" + str(to_disable))
+        if changed:
+            args="subscription-manager repos "
+            for n in to_disable:
+                args += " --disable=" + n
+
+            for n in to_enable:
+                args+=" --enable=" + n
+
+            notice("redhat_repository: enable_only repos args: " + str(args))
+            rc, stdout, stderr = self.module.run_command(args, check_rc=True)
+
+        return [ changed, dict(enabled=to_enable, disabled=to_disable) ]
+
+    def enablerepo(self, id):
+        if len(id) > 0:
+            args="subscription-manager repos "
+            for n in id:
+                args+=" --enable=" + n
+            notice("redhat_repository: enable repos args: " + str(args))
+            rc, stdout, stderr = self.module.run_command(args, check_rc=True)
+            return True
+        else:
+            return False
+
+    def disablerepo(self, id):
+        if len(id) > 0:
+            args="subscription-manager repos "
+            for n in id:
+                args += " --disable=" + n
+            rc, stdout, stderr = self.module.run_command(args, check_rc=True)
+            return True
+        else:
+            return False
+
+    def disable_all(self):
+        notice("redhat_repository: disable all repos")
+        args = "subscription-manager repos --disable='*'"
+        rc, stdout, stderr = self.module.run_command(args, check_rc=True)
+
+
+def main():
+    #
+    # Initialise Ansible module
+    #
+    module = AnsibleModule(
+        argument_spec = dict(
+            id      = dict(type="list"),
+            # can't set a default='all' on list, otherwise id will never be used to enable/disable repos
+            list    = dict(choices=['all', 'enabled', 'disabled']),
+            mode    = dict(choices=['idempotent', 'incremental']),
+            state   = dict(choices=['enabled', 'disabled']),
+        ),
+        required_one_of = [['id', 'list']],
+        mutually_exclusive = [['id', 'list']],
+        supports_check_mode = False,
+    )
+
+    #
+    # Initialize RhsmRepositories
+    #
+    rhsmrepos = RhsmRepositories(module)
+
+    p_list = module.params['list']
+    p_id = module.params['id']
+    p_state = module.params['state']
+    p_mode = module.params['mode']
+
+    # setting the default here, because if done at the AnsibleModule declaration, it screws things up
+    if p_mode is None:
+        p_mode='idempotent'
+
+    notice( "redhat_repository: in main | list=" + str(p_list) + " | id=" + str(p_id) + " | state=" + str(p_state) + " | mode=" + str(p_mode) )
```

Comment:
This line is also too long even by the Ansible code style guide. It should be under 120 chars per line. Please format it like this:

```
notice(
    "redhat_repository: in main | list=" + str(p_list) + " | id=" + str(p_id) + " | state=" + 
    str(p_state) + " | mode=" + str(p_mode))
```